app/src/main/res/layout/submitted_answer_item.xml

```diff
@@ -56,6 +56,7 @@
       android:layout_width="wrap_content"
       android:layout_height="wrap_content"
       android:background="@drawable/submitted_answer_background"
+      android:maxWidth="280dp"
```

Comment:
There is a big variety in the android screen sizes, so, we can't use fixed values.

Reply:
@MohamedMedhat1998 Should we try to first calculate available space programmatically and then try to set maxWidth to say 80% of the available space?

Reply:
I remember trying to do that during GSoC but I was busy making a lot of other stuff, so I didn't put much time in debugging this solution. I can't remember whether it didn't work or I didn't try it to the end, but you can try it.

Reply:
Rather than going with a custom programmatic solution, could we try solving this with existing layouts? ConstraintLayout seems like it should be able to do this: https://stackoverflow.com/a/37369672.

@aggarwalpulkit596 in case you have any thoughts.

Reply:
> Rather than going with a custom programmatic solution, could we try solving this with existing layouts? ConstraintLayout seems like it should be able to do this: https://stackoverflow.com/a/37369672.
> 
> @aggarwalpulkit596 in case you have any thoughts.

@BenHenning This link suggests the use of guidelines, I already tried guidelines to resolve this but then we won't be able to retain the "adjustable width feature" of ```submitted_answer_textview```. Please correct me if I am wrong.

Reply:
I also thought of using ```layout_weight``` but that also offers the same problem, that would make the width of the answer_text strictly to the specified value.

Reply:
Hmm thinking about this more & after finding https://stackoverflow.com/a/46833711, could we do something like:

- LinearLayout -> ConstraintLayout (layout width is match_parent, height is wrap_content same as the current layout)
  - TextView submitted_answer_text_view
    - width -> wrap_content
    - constraint start to parent
    - constraint stop to a guideline that is offset from the parent's edge by the size of the checkmark (should be moved to a dimen)
    - layout_constrainedWidth -> true
  - Checkbox ImageView is positioned relative to the end of the submitted_answer_text_view

The resulting behavior should be that the text grows with the checkbox always to the right of it until the checkbox would be pushed outside the parent at which point the text view should start to start wrapping the text.

Note that this requires ConstraintLayout 1.1 or newer.

Reply:
I think this solution might work as expected because if i remember correctly this i did explore this problem at that time @PrarabdhGarg if it doesn't work let me know I'll give it a try.

Reply:
>     * constraint stop to a guideline that is offset from the parent's edge by the size of the checkmark (should be moved to a dimen)


@BenHenning @aggarwalpulkit596 I am only left with this point, I couldn't find any constraint similar to stop, can you please mention the exact attribute name if possible.

Reply:
@BenHenning I tried fixing it by following other points you mentioned.

Reply:
Ah, I worded that poorly. I meant an end constraint, so something like: ``layout_constraintEnd_toStartOf`` pointing at the barrier.

Does this help clarify?

Reply:
@BenHenning I had a thought that maybe you meant ```layout_constraintEnd_toStartOf``` but the problem is if we do that, and suppose our textview has isn't wide enough then it will be embedded in the middle of the two - 

|            ___________                 |       |
|<--------|___________|----------->|(tick)|  (second last vertical line being the guideline)
|                                                   |       |

Reply:
@BenHenning What are your thoughts?
Also please take a look at the changes I pushed, and note that I've updated the screenshot in the description of the PR

Reply:
Resolving this comment because we have already achieved the desired UI using @BenHenning solution